ui/src/app/utils/generateCheckboxHandlers.ts

```diff
@@ -0,0 +1,25 @@
+import { someInArray } from "./someInArray";
+
+type CheckboxHandlers = {
+  handleGroupCheckbox: (ids: ID[], selectedIDs: ID[]) => void;
+  handleRowCheckbox: (rowID: ID, selectedIDs: ID[]) => void;
+};
+
+type ID = number | string;
+
+export const generateCheckboxHandlers = (
+  onChange: (newSelectedIDs: ID[]) => void
+): CheckboxHandlers => ({
+  handleGroupCheckbox: (ids, selectedIDs) => {
```

Comment:
I reckon a few comments here would really help the next person who needs to modify this.

Reply:
Good call. I missed a JSDoc block too, oops!